### PR TITLE
[Admin-UI-5/7] Warn before self-delete and auto-logout afterwards

### PR DESCRIPTION
**Pull request by PattaraS**
## 🥞 Stacked PR
Use this [link](https://github.com/mlflow/mlflow/pull/22932/files/d202e837496726179eb02ee0635e638c38486ee5..6f95157b85454f4658b3b3c32532038583dfc438) to review incremental changes.
- [stack/fe/rbac/01-backend-auth](https://github.com/mlflow/mlflow/pull/22928) [[Files changed](https://github.com/mlflow/mlflow/pull/22928/files)] [MERGED]
  - [stack/fe/rbac/02-account](https://github.com/mlflow/mlflow/pull/22973) [[Files changed](https://github.com/mlflow/mlflow/pull/22973/files)]
    - [stack/fe/rbac/02-admin-ui-foundation](https://github.com/mlflow/mlflow/pull/22929) [[Files changed](https://github.com/mlflow/mlflow/pull/22929/files/68bca54e3fa479b96764060a31b102af835fa3d7..c0be60d57d847e2ee376c2260d8cc4427ec72abf)]
      - [stack/fe/rbac/04-bulk-delete-and-user-permissions](https://github.com/mlflow/mlflow/pull/22931) [[Files changed](https://github.com/mlflow/mlflow/pull/22931/files/c0be60d57d847e2ee376c2260d8cc4427ec72abf..d202e837496726179eb02ee0635e638c38486ee5)]
        - [**stack/fe/rbac/05-self-delete-confirmation**](https://github.com/mlflow/mlflow/pull/22932) [[Files changed](https://github.com/mlflow/mlflow/pull/22932/files/d202e837496726179eb02ee0635e638c38486ee5..6f95157b85454f4658b3b3c32532038583dfc438)]
          - [stack/fe/rbac/06-dev-user-switcher](https://github.com/mlflow/mlflow/pull/22933) [[Files changed](https://github.com/mlflow/mlflow/pull/22933/files/6f95157b85454f4658b3b3c32532038583dfc438..366ea2d229a6291933a58787801aec67969c5b3b)]

---------

### Related Issues/PRs

Relates to #22724 — Slice 5/6 of #22724. Small dedicated PR for the self-delete UX safety net.

### What changes are proposed in this pull request?

When the user about to be deleted matches the currently authenticated username — either the per-row delete or any row in a bulk-delete selection — surface a warning Alert in the confirmation modal:

> "This is your own account. You'll be logged out immediately after the deletion completes."

After the mutation succeeds, navigate to the prefix-aware `/logout` URL so the browser's Basic Auth cache actually gets cleared instead of leaving the session in a now-broken state. Bulk delete only fires the logout when the self-row's mutation actually succeeded — if every delete failed the user is still authenticated and stays on the page.

The helper `logoutAfterSelfDelete()` resolves `logout` against `window.location.href` so static-prefix deployments keep working. Mirrors the Account-page Logout flow.

Files: `AdminPage.tsx` (delta), i18n + componentId registry deltas (`admin.users.delete_self_warning`, `admin.users.bulk_delete_self_warning`).

#### Stack dependency

Built on top of #22931 (which adds bulk delete). The DevUserSwitcher slice (#22933) lands above this.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

https://github.com/user-attachments/assets/30fb278b-6b5a-499f-b3da-467366530194

### Does this PR require documentation update?

- [x] No.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

### Does this PR require updating the [MLflow Skills](https://github.com/mlflow/skills) repository?

- [x] No.
- [ ] Yes. Please link the corresponding PR or explain how you plan to update it.

#### Is this a user-facing change?

- [ ] No.
- [x] Yes. When an admin deletes their own account from the admin UI (single-row or bulk), a warning Alert appears in the confirmation modal and the page navigates to `/logout` after the deletion succeeds.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/prompts`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/projects`: MLproject format, project running backends
- [x] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Is this PR a critical bugfix or security fix that should go into the next patch release?

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Minor releases are expected to contain larger changes, such as new features and improvements. Non-critical bug fixes and doc updates can be included as well. By default, your PR should target the next minor release.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Patch releases are typically only performed when there has been a major regression or bug in the latest release. For the sake of stability, your PR should not be included in a patch release unless it is a critical fix, or if the risk level of your PR is exceedingly low.

</details>

- [ ] This PR is critical and needs to be in the next patch release
- [x] This PR can wait for the next minor release

### CODE REVIEW

**Comment by PattaraS** [comment]
/review

---

✅ [Review](https://github.com/mlflow/mlflow/actions/runs/24999697672?pr=22932) completed (131.8s, 11 turns, 3754 tokens, $0.26)
<details><summary>Review Output</summary>

PR #22932 has been approved. The self-delete warning and auto-logout logic is correct, handles edge cases well (bulk-delete only logs out if the self-row actually succeeded), and the component IDs are properly registered. No issues found.

</details>

File: tests/server/auth/test_client.py [hunk 1]
```diff
@@ -41,18 +41,19 @@ def clear_credentials(monkeypatch):
 
 @pytest.fixture
 def client(tmp_path):
-    auth_config_path = write_isolated_auth_config(tmp_path)
+    # clean up users & permissions created from previous tests
+    db_file = os.path.abspath(os.path.basename(auth_config.database_uri))
+    if os.path.exists(db_file):
+        os.remove(db_file)
```
[the remark this answers is not in the document]
Comment (PattaraS) [reply tests/server/auth/test_client.py]: Stale — fixed earlier in #22928 (commit `aed8982f7`), inherited via rebase. Resolving as already addressed.

🤖 *Claude response*

File: mlflow/server/js/src/admin/pages/AdminPage.tsx [hunk 2]
```diff
@@ -0,0 +1,827 @@
...
+      <Modal
+        componentId="admin.users.bulk_delete_modal"
+        title="Delete users"
+        visible={bulkDeleteOpen}
+        onCancel={() => setBulkDeleteOpen(false)}
+        onOk={handleBulkDelete}
+        okText="Delete"
+        okButtonProps={{ danger: true }}
+        confirmLoading={deleteUser.isLoading}
+      >
```
[the remark this answers is not in the document]
Comment (PattaraS) [reply mlflow/server/js/src/admin/pages/AdminPage.tsx:416]: Acknowledged. `confirmLoading={delete*.isLoading}` reflects the latest mutation only and can flicker false during a parallel batch. Tracked alongside the N+1 / bulk-delete refactor follow-up.

🤖 *Claude response*

File: mlflow/server/js/src/admin/pages/AdminPage.tsx [hunk 3]
```diff
@@ -0,0 +1,827 @@
...
+      <Modal
+        componentId="admin.roles.bulk_delete_modal"
+        title="Delete roles"
+        visible={bulkDeleteOpen}
+        onCancel={() => setBulkDeleteOpen(false)}
+        onOk={handleBulkDelete}
+        okText="Delete"
+        okButtonProps={{ danger: true }}
+        confirmLoading={deleteRole.isLoading}
+      >
```
[the remark this answers is not in the document]
Comment (PattaraS) [reply mlflow/server/js/src/admin/pages/AdminPage.tsx:786]: Acknowledged. `confirmLoading={delete*.isLoading}` reflects the latest mutation only and can flicker false during a parallel batch. Tracked alongside the N+1 / bulk-delete refactor follow-up.

🤖 *Claude response*

File: mlflow/server/js/src/admin/pages/AdminPage.tsx [hunk 4]
```diff
@@ -0,0 +1,827 @@
...
+
+// Renders one line per role assigned to a user, formatted as
+// `<workspace> → <role_name>`. Mirrors the `<scope> → <value>` shape used
+// elsewhere in the admin UI (e.g. the Account page's permission lines like
+// `experiment:* → READ`). Each row issues its own request — React Query
+// caches per-username so subsequent re-renders don't re-fetch.
+const UserRolesCell = ({ username }: { username: string }) => {
+  const { theme } = useDesignSystemTheme();
+  const { data, isLoading } = useUserRolesQuery(username);
+  if (isLoading) {
```
[the remark this answers is not in the document]
Comment (PattaraS) [reply mlflow/server/js/src/admin/pages/AdminPage.tsx:55]: Acknowledged but deferring — same N+1 / batch-endpoint follow-up tracked under #22929. The user count in the admin UI is small in practice and React Query caches per-username.

🤖 *Claude response*

File: mlflow/server/auth/__init__.py [hunk 5]
```diff
@@ -3014,6 +3062,62 @@ def alert(href: str):
...
+    # Tighten the inline-script attack surface: ``default-src 'none'`` blocks
+    # everything by default, and we explicitly allow only the inline script
+    # (with ``unsafe-inline`` since it's a one-shot bootstrap that doesn't
+    # benefit from a hash) plus inline ``style`` attributes the body uses.
+    # ``frame-ancestors 'none'`` is the same anti-clickjacking guard set
+    # via ``X-Frame-Options`` above, kept for older browsers.
+    res.headers["Content-Security-Policy"] = (
+        "default-src 'none'; "
+        "script-src 'unsafe-inline'; "
+        "style-src 'unsafe-inline'; "
```
[the remark this answers is not in the document]
Comment (PattaraS) [reply mlflow/server/auth/__init__.py]: Fixed in `1f9b934e4` (CRITICAL — good catch). Added `connect-src 'self'` so the same-origin XHR to `./ajax-api/2.0/mlflow/users/current` is allowed by the CSP. Without this, the cache-clearing trick would silently fail.

🤖 *Claude response*

File: mlflow/server/js/src/admin/hooks.ts [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+ *      somehow set.
+ *   2. The localStorage flag is set:
+ *        localStorage.setItem('mlflow.settings.admin.enable-dev-user-switcher_v1', 'true')
+ *      Mirrors the ``mlflow.settings.telemetry.enable-dev-logging`` pattern.
+ */
+export const ADMIN_ENABLE_DEV_USER_SWITCHER_STORAGE_KEY = 'mlflow.settings.admin.enable-dev-user-switcher';
+
+export const DEV_USER_SWITCHER_ENABLED: boolean =
+  process.env['NODE_ENV'] === 'development' &&
+  getLocalStorageItem(ADMIN_ENABLE_DEV_USER_SWITCHER_STORAGE_KEY, 1, false, false);
```
[the remark this answers is not in the document]
Comment (PattaraS) [reply mlflow/server/js/src/admin/hooks.ts:34]: The fix lives in #22933 (commit `da952413d`/`e5b6dfe77`) where the `_v1`-suffixed constant was applied. #22932 is upstream of #22933 in the stack so it inherits the unfixed value, but landing the stack as a unit (or merging #22933 first) makes them consistent. Resolving since the constant is correct in the head of the stack.

🤖 *Claude response*

File: tests/server/auth/test_client.py [hunk 7]
```diff
@@ -1,6 +1,11 @@
 from contextlib import contextmanager
 
 import pytest
+import requests
+
+# A small timeout keeps a hung server from blocking the test suite indefinitely
+# without being so short it false-fires on a slow CI runner.
+_REQUEST_TIMEOUT_SECONDS = 30
 
 import mlflow
```
[the remark this answers is not in the document]
Comment (PattaraS) [reply tests/server/auth/test_client.py:6]: Fixed in `774a4ecef` (#22929). Moved `_REQUEST_TIMEOUT_SECONDS` below the full import block so all imports stay contiguous. Inherited via rebase on #22931 and #22932.

🤖 *Claude response*

**Comment by PattaraS** [comment]
<details>
<summary>Range-diff: stack/fe/rbac/04-bulk-delete-and-user-permissions (13734d3 -&gt; b15e523)</summary>

> <details>
> <summary>mlflow/server/js/src/admin/pages/AdminPage.tsx</summary>
>
> ````diff
> @@ -29,10 +29,10 @@
>     const [showCreateModal, setShowCreateModal] = useState(false);
>     const [newUsername, setNewUsername] = useState('');
>     const [newPassword, setNewPassword] = useState('');
> -   const handleBulkDelete = async () => {
> -     setError(null);
> -     const targets = Array.from(selectedUsernames);
> -+    const includesSelf = selectedUsernames.has(currentUsername);
> +     // Use the reconciled set so we don't try to delete users who already
> +     // disappeared after a refetch / single-row delete.
> +     const targets = Array.from(visibleSelectedUsernames);
> ++    const includesSelf = visibleSelectedUsernames.has(currentUsername);
>       const results = await Promise.allSettled(targets.map((u) => deleteUser.mutateAsync(u)));
>       const failures = results.filter((r) => r.status === 'rejected') as PromiseRejectedResult[];
>       if (failures.length > 0) {
> @@ -83,13 +83,15 @@
>           confirmLoading={deleteUser.isLoading}
>         >
>  -        <Typography.Text>
> --          Delete {selectedUsernames.size} user{selectedUsernames.size === 1 ? '' : 's'}? This action cannot be undone.
> +-          Delete {visibleSelectedUsernames.size} user{visibleSelectedUsernames.size === 1 ? '' : 's'}? This action
> +-          cannot be undone.
>  -        </Typography.Text>
>  +        <div css={{ display: 'flex', flexDirection: 'column', gap: theme.spacing.md }}>
>  +          <Typography.Text>
> -+            Delete {selectedUsernames.size} user{selectedUsernames.size === 1 ? '' : 's'}? This action cannot be undone.
> ++            Delete {visibleSelectedUsernames.size} user{visibleSelectedUsernames.size === 1 ? '' : 's'}? This action
> ++            cannot be undone.
>  +          </Typography.Text>
> -+          {selectedUsernames.has(currentUsername) && (
> ++          {visibleSelectedUsernames.has(currentUsername) && (
>  +            <Alert
>  +              componentId="admin.users.bulk_delete_self_warning"
>  +              type="warning"
> ````
>
> </details>


<sub>Reproduce locally: `git range-diff 4330f0b..13734d3 8407733..b15e523` | Disable: `git config gitstack.push-range-diff false`</sub>
</details>

File: mlflow/server/js/src/common/components/MlflowSidebar.tsx [hunk 8]
```diff
@@ -411,10 +421,26 @@ export function MlflowSidebar({
...
+              actually use without picking a workspace first. */}
+          {!showNestedSettingsItems && (
             <MlflowSidebarLink
               css={{ paddingBlock: theme.spacing.sm }}
-              to={`${ExperimentTrackingRoutes.getSettingsSectionRoute(SETTINGS_SECTION_GENERAL)}?${SETTINGS_RETURN_TO_PARAM}=${encodeURIComponent(location.pathname + location.search)}`}
+              to={
+                showWorkspaceMenuItems
+                  ? `${ExperimentTrackingRoutes.getSettingsSectionRoute(SETTINGS_SECTION_GENERAL)}?${SETTINGS_RETURN_TO_PARAM}=${encodeURIComponent(location.pathname + location.search)}`
+                  : AdminRoutes.accountPageRoute
+              }
```
[the remark this answers is not in the document]
Comment (PattaraS) [reply mlflow/server/js/src/common/components/MlflowSidebar.tsx]: Fixed in `1aef817fd`. Added an `enabled?: boolean` option to `useCurrentUserQuery` and gated the strict caller with `enabled: !showWorkspaceMenuItems` per your suggestion, so the request only fires on the workspace-selection page where `isAuthAvailableStrict` is actually consulted. Other call sites (`useCurrentUserIsAdmin`, `useIsAuthAvailable`) keep firing unconditionally because they're consumed by the always-rendered admin gate — that 404 cost is tracked separately on #22929.

🤖 *Claude response*

**Comment by PattaraS** [comment]
<details>
<summary>Range-diff: stack/fe/rbac/04-bulk-delete-and-user-permissions (c5ee18f -&gt; b56f5e5)</summary>

> <details>
> <summary>mlflow/server/js/src/admin/pages/AdminPage.tsx</summary>
>
> ````diff
> @@ -5,9 +5,9 @@
>     useDeleteRole,
>     useUserRolesQuery,
>  +  useCurrentUserQuery,
> +   useWithSettingsReturnTo,
>   } from '../hooks';
>   import type { CreateRoleRequest } from '../types';
> - import { isWorkspaceAdminRole } from '../types';
>   const UsersTab = () => {
>     const { theme } = useDesignSystemTheme();
>     const { data: usersData, isLoading, error: queryError } = useUsersQuery();
> @@ -16,6 +16,7 @@
>     const createUser = useCreateUser();
>     const deleteUser = useDeleteUser();
>     const updateAdmin = useUpdateAdmin();
> +   const withReturnTo = useWithSettingsReturnTo();
>   
>  +  // Deleting your own account is allowed (an admin removing their own
>  +  // account before someone else takes over is a real flow), but it has the
> ````
>
> </details>


<sub>Reproduce locally: `git range-diff 520cc69..c5ee18f 9249fad..b56f5e5` | Disable: `git config gitstack.push-range-diff false`</sub>
</details>

**Comment by PattaraS** [comment]
<details>
<summary>Range-diff: stack/fe/rbac/04-bulk-delete-and-user-permissions (97fdc5a -&gt; c84c7f9)</summary>

> <details>
> <summary>mlflow/server/js/src/MlflowRouter.tsx</summary>
>
> ````diff
> @@ -1,36 +0,0 @@
> -diff --git a/mlflow/server/js/src/MlflowRouter.tsx b/mlflow/server/js/src/MlflowRouter.tsx
> ---- a/mlflow/server/js/src/MlflowRouter.tsx
> -+++ b/mlflow/server/js/src/MlflowRouter.tsx
> -   const { workflowType } = useWorkflowType();
> - 
> -   return (
> --    <div css={{ display: 'flex', flexDirection: 'column', height: '100%' }}>
> -+    <div
> -+      css={{
> -+        display: 'flex',
> -+        flexDirection: 'column',
> -+        height: '100%',
> -+        // Single chrome paint surface for the whole viewport. Both the
> -+        // top-bar and the sidebar/main row are transparent above this so
> -+        // the gradient is continuous across the seam between them — each
> -+        // layer painting its own gradient would compute the pattern
> -+        // against its own bounding box and the bands wouldn't match.
> -+        background:
> -+          workflowType === WorkflowType.GENAI
> -+            ? `linear-gradient(163deg, rgba(66, 153, 224, 0.06) 20%, rgba(202, 66, 224, 0.06) 35%, rgba(255, 95, 70, 0.06) 50%, transparent 80%), ${theme.colors.backgroundSecondary}`
> -+            : theme.colors.backgroundSecondary,
> -+      }}
> -+    >
> -       <ErrorModal />
> -       <AppErrorBoundary>
> -         <MlflowTopBar />
> -               display: 'flex',
> -               flexDirection: 'row',
> -               width: '100%',
> --              background:
> --                workflowType === WorkflowType.GENAI
> --                  ? `linear-gradient(163deg, rgba(66, 153, 224, 0.06) 20%, rgba(202, 66, 224, 0.06) 35%, rgba(255, 95, 70, 0.06) 50%, transparent 80%), ${theme.colors.backgroundSecondary}`
> --                  : theme.colors.backgroundSecondary,
> -             }}
> -           >
> -             <MlflowSidebar showSidebar={showSidebar} setShowSidebar={setShowSidebar} />
> \ No newline at end of file
> ````
>
> </details>

> <details>
> <summary>mlflow/server/js/src/common/components/MlflowTopBar.tsx</summary>
>
> ````diff
> @@ -1,20 +0,0 @@
> -diff --git a/mlflow/server/js/src/common/components/MlflowTopBar.tsx b/mlflow/server/js/src/common/components/MlflowTopBar.tsx
> ---- a/mlflow/server/js/src/common/components/MlflowTopBar.tsx
> -+++ b/mlflow/server/js/src/common/components/MlflowTopBar.tsx
> -  * avatar (Databricks "circle + first character" pattern) opens a dropdown
> -  * with Account and Logout. Renders an empty shell when auth is unavailable
> -  * so the layout doesn't shift between auth-on / auth-off MLflow deployments.
> -+ *
> -+ * The top-bar is intentionally transparent — the chrome background lives on
> -+ * the outer ``MlflowRootLayout`` wrapper so the gradient (in GenAI mode)
> -+ * paints continuously across the top-bar and the sidebar/main row below.
> -  */
> - export const MlflowTopBar = () => {
> -   const { theme } = useDesignSystemTheme();
> -         justifyContent: 'flex-end',
> -         height: theme.spacing.xl + theme.spacing.md,
> -         paddingInline: theme.spacing.md,
> --        backgroundColor: theme.colors.backgroundSecondary,
> -         gap: theme.spacing.sm,
> -         flexShrink: 0,
> -       }}
> \ No newline at end of file
> ````
>
> </details>

> <details>
> <summary>mlflow/server/js/src/common/components/RootAssistantLayout.tsx</summary>
>
> ````diff
> @@ -1,15 +0,0 @@
> -diff --git a/mlflow/server/js/src/common/components/RootAssistantLayout.tsx b/mlflow/server/js/src/common/components/RootAssistantLayout.tsx
> ---- a/mlflow/server/js/src/common/components/RootAssistantLayout.tsx
> -+++ b/mlflow/server/js/src/common/components/RootAssistantLayout.tsx
> -           display: 'flex',
> -           flexGrow: 1,
> -           minHeight: 0,
> --          backgroundColor: theme.colors.backgroundSecondary,
> -+          // Chrome background lives on the outer ``MlflowRootLayout``
> -+          // wrapper so it paints continuously behind the top-bar and the
> -+          // row beneath it. Stay transparent here so the gradient shows
> -+          // through; the assistant panel itself sets its own
> -+          // ``backgroundPrimary``, so it's unaffected.
> -         }}
> -       >
> -         <div css={{ flex: 1, minWidth: 0, display: 'flex' }}>{children}</div>
> \ No newline at end of file
> ````
>
> </details>


<sub>Reproduce locally: `git range-diff 7a1e753..97fdc5a 1e3beb6..c84c7f9` | Disable: `git config gitstack.push-range-diff false`</sub>
</details>

**Comment by PattaraS** [comment]
<details>
<summary>Range-diff: stack/fe/rbac/04-bulk-delete-and-user-permissions (7979227 -&gt; 6c5e991)</summary>

> <details>
> <summary>mlflow/server/js/src/admin/pages/AdminPage.tsx</summary>
>
> ````diff
> @@ -1,6 +1,17 @@
>  diff --git a/mlflow/server/js/src/admin/pages/AdminPage.tsx b/mlflow/server/js/src/admin/pages/AdminPage.tsx
>  --- a/mlflow/server/js/src/admin/pages/AdminPage.tsx
>  +++ b/mlflow/server/js/src/admin/pages/AdminPage.tsx
> + } from '@databricks/design-system';
> + import { FormattedMessage } from 'react-intl';
> + import { ScrollablePageWrapper } from '@mlflow/mlflow/src/common/components/ScrollablePageWrapper';
> ++import { useQueryClient } from '@mlflow/mlflow/src/common/utils/reactQueryHooks';
> + import { Link, useSearchParams } from '../../common/utils/RoutingUtils';
> + import { useWorkspaces } from '../../workspaces/hooks/useWorkspaces';
> + import { useWorkspacesEnabled } from '../../experiment-tracking/hooks/useServerInfo';
> ++import { performLogout } from '../auth-utils';
> + import AdminRoutes from '../routes';
> + import {
> +   useUsersQuery,
>     useCreateRole,
>     useDeleteRole,
>     useUserRolesQuery,
> @@ -8,8 +19,10 @@
>     useWithSettingsReturnTo,
>   } from '../hooks';
>   import type { CreateRoleRequest } from '../types';
> + 
>   const UsersTab = () => {
>     const { theme } = useDesignSystemTheme();
> ++  const queryClient = useQueryClient();
>     const { data: usersData, isLoading, error: queryError } = useUsersQuery();
>  +  const { data: currentUserData } = useCurrentUserQuery();
>  +  const currentUsername = currentUserData?.user?.username ?? '';
> @@ -23,9 +36,7 @@
>  +  // side effect of logging you out — surface that explicitly in the
>  +  // confirmation, and follow it through after the deletion succeeds so
>  +  // the browser doesn't sit on a now-broken auth state.
> -+  const logoutAfterSelfDelete = () => {
> -+    window.location.href = new URL('logout', window.location.href).toString();
> -+  };
> ++  const logoutAfterSelfDelete = () => performLogout(queryClient);
>  +
>     const [showCreateModal, setShowCreateModal] = useState(false);
>     const [newUsername, setNewUsername] = useState('');
> ````
>
> </details>


<sub>Reproduce locally: `git range-diff 495fea5..7979227 c6dfdc7..6c5e991` | Disable: `git config gitstack.push-range-diff false`</sub>
</details>

**Comment by PattaraS** [comment]
<details>
<summary>Range-diff: stack/fe/rbac/04-bulk-delete-and-user-permissions (2e62ab0 -&gt; c2980e4)</summary>

> <details>
> <summary>mlflow/server/js/src/admin/pages/AdminPage.tsx</summary>
>
> ````diff
> @@ -12,20 +12,11 @@
>   import AdminRoutes from '../routes';
>   import {
>     useUsersQuery,
> -   useCreateRole,
> -   useDeleteRole,
> -   useUserRolesQuery,
> -+  useCurrentUserQuery,
> -   useWithSettingsReturnTo,
> - } from '../hooks';
> - import type { CreateRoleRequest } from '../types';
>   
>   const UsersTab = () => {
>     const { theme } = useDesignSystemTheme();
>  +  const queryClient = useQueryClient();
>     const { data: usersData, isLoading, error: queryError } = useUsersQuery();
> -+  const { data: currentUserData } = useCurrentUserQuery();
> -+  const currentUsername = currentUserData?.user?.username ?? '';
>     const createUser = useCreateUser();
>     const deleteUser = useDeleteUser();
>     const updateAdmin = useUpdateAdmin();
> @@ -40,79 +31,4 @@
>  +
>     const [showCreateModal, setShowCreateModal] = useState(false);
>     const [newUsername, setNewUsername] = useState('');
> -   const [newPassword, setNewPassword] = useState('');
> -     // Use the reconciled set so we don't try to delete users who already
> -     // disappeared after a refetch / single-row delete.
> -     const targets = Array.from(visibleSelectedUsernames);
> -+    const includesSelf = visibleSelectedUsernames.has(currentUsername);
> -     const results = await Promise.allSettled(targets.map((u) => deleteUser.mutateAsync(u)));
> -     const failures = results.filter((r) => r.status === 'rejected') as PromiseRejectedResult[];
> -     if (failures.length > 0) {
> -     }
> -     setSelectedUsernames(new Set());
> -     setBulkDeleteOpen(false);
> -+    // Only log out if the self-delete actually succeeded (not in the
> -+    // failures list). If every delete failed, the user is still
> -+    // authenticated — leave them on the page.
> -+    if (includesSelf) {
> -+      const selfResult = results[targets.indexOf(currentUsername)];
> -+      if (selfResult?.status === 'fulfilled') {
> -+        logoutAfterSelfDelete();
> -+      }
> -+    }
> -   };
> - 
> -   const handleCreateUser = async () => {
> -   const handleDeleteUser = async () => {
> -     if (!deleteTarget) return;
> -     setError(null);
> -+    const isSelf = deleteTarget === currentUsername;
> -     try {
> -       await deleteUser.mutateAsync(deleteTarget);
> -       setDeleteTarget(null);
> -+      if (isSelf) {
> -+        logoutAfterSelfDelete();
> -+      }
> -     } catch (e: any) {
> -       setError(e.message || 'Failed to delete user');
> -     }
> -           <Typography.Text>
> -             Are you sure you want to delete user <strong>{deleteTarget}</strong>? This action cannot be undone.
> -           </Typography.Text>
> -+          {deleteTarget === currentUsername && (
> -+            <Alert
> -+              componentId="admin.users.delete_self_warning"
> -+              type="warning"
> -+              message="This is your own account."
> -+              description="You'll be logged out immediately after the deletion completes."
> -+              closable={false}
> -+            />
> -+          )}
> -         </div>
> -       </Modal>
> -       <Modal
> -         okButtonProps={{ danger: true }}
> -         confirmLoading={deleteUser.isLoading}
> -       >
> --        <Typography.Text>
> --          Delete {visibleSelectedUsernames.size} user{visibleSelectedUsernames.size === 1 ? '' : 's'}? This action
> --          cannot be undone.
> --        </Typography.Text>
> -+        <div css={{ display: 'flex', flexDirection: 'column', gap: theme.spacing.md }}>
> -+          <Typography.Text>
> -+            Delete {visibleSelectedUsernames.size} user{visibleSelectedUsernames.size === 1 ? '' : 's'}? This action
> -+            cannot be undone.
> -+          </Typography.Text>
> -+          {visibleSelectedUsernames.has(currentUsername) && (
> -+            <Alert
> -+              componentId="admin.users.bulk_delete_self_warning"
> -+              type="warning"
> -+              message="Your own account is in this selection."
> -+              description="You'll be logged out immediately after the deletion completes."
> -+              closable={false}
> -+            />
> -+          )}
> -+        </div>
> -       </Modal>
> -     </div>
> -   );
> \ No newline at end of file
> +   const [newPassword, setNewPassword] = useState('');
> \ No newline at end of file
> ````
>
> </details>

> <details>
> <summary>.github/actions/check-component-ids/componentId-registry.js</summary>
>
> ````diff
> @@ -1,17 +0,0 @@
> -diff --git a/.github/actions/check-component-ids/componentId-registry.js b/.github/actions/check-component-ids/componentId-registry.js
> ---- a/.github/actions/check-component-ids/componentId-registry.js
> -+++ b/.github/actions/check-component-ids/componentId-registry.js
> -   "admin.users.admin_header": "",
> -   "admin.users.bulk_delete_button": "",
> -   "admin.users.bulk_delete_modal": "",
> -+  "admin.users.bulk_delete_self_warning": "",
> -   "admin.users.create_button": "",
> -   "admin.users.create_modal": "",
> -   "admin.users.create_modal.error": "",
> -   "admin.users.delete_button": "",
> -   "admin.users.delete_modal": "",
> -   "admin.users.delete_modal.error": "",
> -+  "admin.users.delete_self_warning": "",
> -   "admin.users.error": "",
> -   "admin.users.query_error": "",
> -   "admin.users.roles_header": "",
> \ No newline at end of file
> ````
>
> </details>


<sub>Reproduce locally: `git range-diff 652155a..2e62ab0 bfd1093..c2980e4` | Disable: `git config gitstack.push-range-diff false`</sub>
</details>

**Comment by PattaraS** [comment]
Folded into #22931 — the logoutAfterSelfDelete helper is a 10-line addition to AdminPage.tsx, which #22931 already restructures for bulk-delete. Combining keeps related delete-flow logic in one PR. Closing.
